clippy_lints/src/neg_multiply.rs

```diff
@@ -52,7 +52,11 @@ fn check_mul(cx: &LateContext<'_>, span: Span, lit: &Expr<'_>, exp: &Expr<'_>) {
     if let ExprKind::Lit(l) = lit.kind
         && matches!(
             consts::lit_to_mir_constant(&l.node, cx.typeck_results().expr_ty_opt(lit)),
-            Constant::Int(1) | Constant::F16(1.0) | Constant::F32(1.0) | Constant::F64(1.0) | Constant::F128(1.0)
+            Constant::Int(1)
+                | Constant::F16(0x3c00)
```

Comment:
Should use `const { 1.0f16.to_bits() }` rather than hard coding the constant. Same for `f128`.